test/Common/TestHelpers.cs

```diff
@@ -192,6 +193,24 @@ internal static class TestHelpers
                 onSend: onSend);
         }
 
+        public static IPlatformInformationService GetMockPlatformInformationService(
```

Comment:
Instead of mocking IPlatformInformationService, it may be better to mock the underlying INameResolver, as that mimics production slightly better.

Considering how slim of a wrapper IPlatformInformationService is it may not make a huge difference, but it's worth considering.

Reply:
That's true, but I'm not sure that we gain that much benefit from the cost of making the mock a little bit more low-level. Let me get back to this suggestion. I'll want to first prioritize the other bits :)

Reply:
Given the timeline, I'm gonna opt to wait until the OOProc performance tuning PR to consider this change.